client/src/components/RegisterPage/Registration.js

```diff
@@ -10,38 +13,40 @@ const styles = makeStyles((theme) => ({
 }));
 
 const Register = (props) => {
-  const [user, setUser] = useState({
+  const [user, setUserInfo] = useState({
     username: "",
     email: "",
     password: "",
     secondPass: "",
   });
 
+  const { setUser } = React.useContext(UserContext);
+
   const handleChange = (e) => {
     const { name, value } = e.target;
     if (name === "username") {
-      setUser({
+      setUserInfo({
         username: value,
         email: user["email"],
         password: user["password"],
         secondPass: user["secondPass"],
       });
     } else if (name === "email") {
-      setUser({
+      setUserInfo({
         username: user["username"],
         email: value,
         password: user["password"],
         secondPass: user["secondPass"],
       });
     } else if (name === "password") {
-      setUser({
+      setUserInfo({
         username: user["username"],
         email: user["email"],
         password: value,
         secondPass: user["secondPass"],
       });
     } else if (name === "secondPass") {
-      setUser({
+      setUserInfo({
         username: user["username"],
         email: user["email"],
         password: user["password"],
```

Comment:
could still use '...user' but works

Reply:
Agreed, when I did this I just changed the name of the function and left everything else as is though.